src/NovelImageRect.cpp

```diff
@@ -2,42 +2,54 @@
 // Created by matth on 05/01/2019.
 //
 
+#include <iostream>
 #include "NovelImageRect.h"
 
 namespace NovelRT {
 
-    NovelImageRect::NovelImageRect(const float& screenScale, const GeoVector<float> &position, NVGcontext *context, const std::string &imageDir,
-                                   const float &rotation, const GeoVector<float>& scale) : NovelObject(screenScale, position, GeoVector<float>(0, 0), context,
-                                                                        rotation, scale), _imageDir(imageDir) {
-        _imageHandle = nvgCreateImage(_context, _imageDir.c_str(), NVG_IMAGE_NEAREST);
-        _width = 0;
-        _height = 0;
-        nvgImageSize(_context, _imageHandle, &_width, &_height);
-        setSize(GeoVector<float>(_width, _height));
-        setScale(scale);
-    }
-
-    void NovelImageRect::setScale(const GeoVector<float>& value)  {
-        _paint = nvgImagePattern(_context, 0, 0, (_width * _screenScale) * value.getX(), (_height * _screenScale) * value.getY(), 0, _imageHandle, 1);
-        NovelObject::_scale = value;
-    }
-
-    void NovelImageRect::drawObject() const {
-        if (!getActive()) return;
-
-        GeoVector<float> position = getPosition() * _screenScale;
-        GeoVector<float> size = getSize() * _screenScale;
-        size = size * getScale();
-        nvgSave(_context);
-        nvgTranslate(_context, position.getX(), position.getY());
-        nvgRotate(_context, getRotation());
-        nvgTranslate(_context, -(size.getX() / 2), -(size.getY() / 2));
-        nvgBeginPath(_context);
-        nvgRect(_context, 0, 0, size.getX(), size.getY());
-        nvgFillPaint(_context, _paint);
-        nvgFill(_context);
-        nvgRestore(_context);
-    }
-
-
-}
\ No newline at end of file
+NovelImageRect::NovelImageRect(NovelLayeringService* layeringService,
+                               const float& screenScale,
+                               const std::string_view imageDir,
+                               const NovelCommonArgs& args,
+                               NovelRenderingService* renderingService) :
+    NovelRenderObject(layeringService, screenScale, GeoVector<float>(0, 0), args, renderingService),
+    _imageDir(imageDir) {
+
+  _imageHandle = nvgCreateImage(_drawContext, _imageDir.c_str(), NVG_IMAGE_NEAREST);
+  _width = 0;
+  _height = 0;
+  nvgImageSize(_drawContext, _imageHandle, &_width, &_height);
+  setSize(GeoVector<float>(_width, _height));
+  setScale(args.startingScale);
+}
+
+void NovelImageRect::setScale(const GeoVector<float>& value) {
+  _paint = nvgImagePattern(_drawContext,

```

Comment:
this style seems pretty inconsistent with other long parameter lines - not sure if this is how it is supposed to be, however

Reply:
Yeah not worth fixing - nanovg will be gone come my next branch.